tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -281,60 +278,112 @@ private void SetRemoteConfigCapabilites()
 
         private void AsmDDProductConfigChanged(object sender, ProductConfigChangedEventArgs e)
         {
-            var asmDD = e.GetDeserializedConfigurations<string>().FirstOrDefault();
-            if (!string.IsNullOrEmpty(asmDD.TypedFile))
+            lock (_settings)
             {
-                _remoteRulesJson = asmDD.TypedFile;
-                UpdateStatus(true);
-            }
+                var asmDD = e.GetDeserializedConfigurations<string>().FirstOrDefault();
+                if (!string.IsNullOrEmpty(asmDD.TypedFile))
+                {
+                    _remoteRulesJson = asmDD.TypedFile;
+                    UpdateStatus(true);
+                }
 
-            e.Acknowledge(asmDD.Name);
+                e.Acknowledge(asmDD.Name);
+            }
         }
 
         private void FeaturesProductConfigChanged(object sender, ProductConfigChangedEventArgs e)
         {
-            var features = e.GetDeserializedConfigurations<AsmFeatures>().FirstOrDefault();
-            if (features.TypedFile != null)
+            lock (_settings)
```

Comment:
Not sure I understand why this lock is necessary and why it's needed across such a large scope in this method.

Also, why lock on `_settings`? It appears to be used protect several different fields in this project. Using a separate field for the lock would make things clearer, I think.

Reply:
It's just to sync changes to config globally. But I can fine tune it.

Reply:
Fixed in 30e6e8d6aade2c374900d70a7e45cea9a948beca